pkg/catalog-cd/fetcher/fetcher.go

```diff
@@ -0,0 +1,172 @@
+package fetcher
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/go-gh/v2/pkg/api"
+	"github.com/openshift-pipelines/tektoncd-catalog/pkg/catalog-cd/fetcher/config"
+)
+
+// TODO: prepare the "dest" workspace
+//
+//	(fetch the repository's `p` branch, …)
+//
+// TODO: fetch release assets
+//   - fetch yamls
+//   - fetch tests (yamls with kttl)
```

Comment:
Testing should be managed by `catalog-cd probe`, therefore the settings needed for testing should take part of the contract itself. At least initially I think we should narrow down the type of Kubernetes resources.

Reply:
yeah those comments are "old".. but I kept them to think about the "test" part of the contract indeed.